--- diffexampleOld.js	2023-09-19 10:20:06.000000000 -0400
+++ diffexampleNew.js	2023-09-20 22:18:24.000000000 -0400
@@ -8,35 +8,40 @@
 that natively without any external plugins
 */
 
-//Node JS dependencies for reading and writing files to the file system.
+//Node JS dependencies for reading and writing files to the file system. --revised here
 var fs = require('fs');
-//Read the file from the file system
-var obj = JSON.parse(fs.readFileSync('./data.json','utf8'));
-//get the date from the first item.
-var date = new Date(obj[0].timestamp);
-//format the date to the YYYYMMDD.csv required for the submission.
-var datef = date.getFullYear() + pad2(date.getMonth()+1) + pad2(date.getDate());
+var jsondata = process.argv[2];
+
+//Read and parse the file from the file system --revised here
+var obj = fs.readFileSync(jsondata);
+const jsonparsing = JSON.parse(obj);
+
+//format the date to the YYYYMMDD.csv required for the submission. --revised here
+var dt = Date.now()
+var datef = new Date(dt)
+
 //to return string object
 var csv = ""
+
 //for loop to parse all items
-for(var i = 0; i < obj.length; i++){
+for(var i = 0; i < jsonparsing.length; i++){
   //item is a single person
-  var item = obj[i];
-  //check if item has a creditcard then save the :name and :creditcard to the cvs object in the cvs format
-  if(item.creditcard != null){
-    csv += item.name + "," + item.creditcard + "\n";
+  var item = jsonparsing[i];
+  //check if item has a creditcard and email then save the :name and :creditcard and :email to the csv object in the csv format --revised here
+  if(item.email != null && item.creditcard != null){
+    csv += item.name + "," + item.email + "," + item.creditcard + "\n";
   }
 }
 
-//store the compiled cvs in a file formated YYYYMMDD.csv
-fs.writeFile("./" + datef + ".csv", csv, function(err) {
+//store the compiled csv in a file formated YYYYMMDD.csv --revised here
+fs.writeFile("./" + datef.getFullYear() + addzero(datef.getMonth() + 1) + addzero(datef.getDate()) + ".csv", csv, function(err) {
     if(err) {
         return console.log(err);
     }
     console.log("The file was saved!");
 });
 
-//formating functino for manking numbebers 2 digits ex "2" -> "02"
-function pad2(number) {
+//formating function for making numbers 2 digits ex "2" -> "02" --revised here
+function addzero(number) {
      return (number < 10 ? '0' : '') + number
-}
+}
\ No newline at end of file
